SleepEngine/Engine/Components/Component.h

```diff
@@ -26,6 +38,8 @@ class Component
 protected:
     NotOwnedPtr <Object> m_parent{ nullptr };
 
+    virtual void onAddedAsParent() {}
```

Comment:
Let's create ComponentWithHierarchy to contain parentComponent and this method? Because most parts of components don't need it.
Or let's store parentComponent in every Component if you wish, but I guess it's better to create a different class for this